src/screens/OrganizationEvents/CustomRecurrenceModal.tsx

```diff
@@ -0,0 +1,276 @@
+import React, { useEffect, useState } from 'react';
+import { Button, Dropdown, Form, FormControl, Modal } from 'react-bootstrap';
+import styles from './OrganizationEvents.module.css';
+import { DatePicker } from '@mui/x-date-pickers';
+import { Days, Frequency, frequencies } from 'utils/recurrenceRuleUtils';
+import type {
+  InterfaceRecurrenceRule,
+  WeekDays,
+} from 'utils/recurrenceRuleUtils';
+import type { Dayjs } from 'dayjs';
+import dayjs from 'dayjs';
+
+const daysOptions = ['S', 'M', 'T', 'W', 'T', 'F', 'S'];
+const endsOnOptions = ['never', 'on', 'after'];
+
+interface InterfaceCustomRecurrenceModalProps {
+  recurrenceRuleState: InterfaceRecurrenceRule;
+  setRecurrenceRuleState: (
+    state: React.SetStateAction<InterfaceRecurrenceRule>,
+  ) => void;
+  endDate: Date | null;
+  setEndDate: (state: React.SetStateAction<Date | null>) => void;
+  customRecurrenceModalIsOpen: boolean;
+  hideCustomRecurrenceModal: () => void;
+  setCustomRecurrenceModalIsOpen: (
+    state: React.SetStateAction<boolean>,
+  ) => void;
+  t: (key: string) => string;
+}
+
+const CustomRecurrenceModal: React.FC<InterfaceCustomRecurrenceModalProps> = ({
+  recurrenceRuleState,
+  setRecurrenceRuleState,
+  endDate,
+  setEndDate,
+  customRecurrenceModalIsOpen,
+  hideCustomRecurrenceModal,
+  setCustomRecurrenceModalIsOpen,
+  t,
+}) => {
+  const { frequency, weekDays } = recurrenceRuleState;
+  const [endsOnOption, setEndsOnOption] = useState<string>('never');
+
+  useEffect(() => {
+    if (endDate) {
+      setEndsOnOption('on');
+    }
+  }, [endDate]);
+
+  const handleEndsOnOptionChange = (
+    e: React.ChangeEvent<HTMLInputElement>,
+  ): void => {
+    const endsOnOption = e.target.value;
+    setEndsOnOption(endsOnOption);
+    if (endsOnOption === 'never') {
```

Comment:
Can these states such as 'never', 'on', 'after' be placed in constants and referenced throughout the code. This will help to restrict issues of misspellings.

https://dev.to/ruben_alapont/magic-numbers-and-magic-strings-its-time-to-talk-about-it-ci2#:~:text=In%20the%20world%20of%20JavaScript,as%20it%20evolves%20over%20time.

Reply:
Ahh right! I forgot to add a type here. I'll do that, thanks.

'never', 'on' and 'after' here are options for specifying the end of recurrence, which is an enum type.